### ISSUE TITLE
Cannot use import statement outside a module

### ISSUE THREAD

**Issue by colinricardo** [JohannesKlauss/react-hotkeys-hook#795]
Encountering this on all v4 releases:

<img width="496" alt="CleanShot 2022-08-23 at 21 41 36@2x" src="https://user-images.githubusercontent.com/21294951/186261488-add4a20f-8687-41e3-8672-c7c3cf2ac041.png">

along with a lot of output: 

<img width="938" alt="CleanShot 2022-08-23 at 21 42 11@2x" src="https://user-images.githubusercontent.com/21294951/186261524-dda36990-3015-4c66-b856-136a3efd2a3c.png">

**Comment by JohannesKlauss** [comment]
How are you importing the Module?

**Comment by JohannesKlauss** [comment]
Version `4.0.0-5` seems to work just fine: https://codesandbox.io/s/thirsty-mopsa-x43bfw?file=/src/App.tsx

**Comment by JohannesKlauss** [comment]
If this issue persists, please provide new info, for now I am closing it due to inactivity.

**Comment by tmcw** [comment]
Seeing this on v4.0.2, importing via Next.js 12.3.1 - looks like the default distributed module changed from CJS to ESM between 3 & 4. Which is fine, but might want to highlight "ESM only" because this'll break a lot of users setups.

**Comment by Rednegniw** [comment]
Seeing this on my setup as well, with Next.js 13.0.2.

**Comment by jkesteris** [comment]
Same here, I'm using 4.0.2 and for nextjs 12.2.5 . here's a link to [codesandbox](https://codesandbox.io/p/sandbox/pedantic-estrela-75zlkr?selection=%5B%7B%22endColumn%22%3A49%2C%22endLineNumber%22%3A6%2C%22startColumn%22%3A49%2C%22startLineNumber%22%3A6%7D%5D&file=%2Fpackage.json&workspace=%257B%2522activeFileId%2522%253A%2522cl8skx9az000nlpgeaw1kc72o%2522%252C%2522openFiles%2522%253A%255B%2522%252FREADME.md%2522%252C%2522%252Fpackage.json%2522%252C%2522%252Fpages%252Findex.tsx%2522%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522sidekickItems%2522%253A%255B%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522key%2522%253A%2522cla3ysmx6006v3b6hafv223j3%2522%252C%2522isMinimized%2522%253Afalse%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522key%2522%253A%2522cla3yskxt005i3b6hr0c9lvqf%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D) that reproduces the issue.

**Comment by JohannesKlauss** [comment]
Thanks for the reproducible, I'll check it out and come back with a fix.

**Comment by nicksrandall** [comment]
I'm also seeing this problem in a Remix app. Thanks for looking into it!

**Comment by ZuBB** [comment]
Also have it 😞 

@JohannesKlauss could you please ping this issue when you release a bugfix for this issue?

**Comment by JohannesKlauss** [comment]
Does it still happen with version 4.0.4-0?

**Comment by lukebowerman** [comment]
FYI - for the Remix folks here... Adding the package to `remix.config.js` `serverDependenciesToBundle` option sorts things out just fine (Remix / ESBuild will then transpile the package from ESM to CJS)

```
  serverDependenciesToBundle: [
    'react-hotkeys-hook',
  ],
```

**Comment by lukebowerman** [comment]
@JohannesKlauss - I just tried out `4.0.4-0` on Remix without transpiling (see my previous comment) I'm getting an error about unexpected output (seems like its barfing on the `dist/index.js`

However running it it through bundler seems to straighten things out and it works fine.

**Comment by lukebowerman** [comment]
Follow-up... turns out our Vitest unit tests are also failing due to this upgrade:

<img width="1290" alt="image" src="https://user-images.githubusercontent.com/19156294/200637947-02bd5f2f-b81f-454b-b2c8-34d280493749.png">

Only change was to bump from `3.4.7` to `4.0.3` and some minor changes to the type interfaces for our use of `useHotkeys`. I'll work to get a minimal reproduction repository up for the Remix issue later today. I suspect Vitest is failing for the same reason (something's not transpiled quite right). 

Thanks so much for you work on this library!

**Comment by MartinXPN** [comment]
Is there a workaround that we can use for now? I'm also using Next.js 13.

**Comment by MartinXPN** [comment]
I'm currently using Next.js 13 with `useHotkeys@4.0.2`.
The weirdest part is that everything works when I have `useHotkeys` in one file. Yet, everything breaks as soon as I add some other `useHotkeys` in another file.

I currently have:
```tsx
// Shortcuts to run and submit code
useHotkeys('meta+s', () => {}, {preventDefault: true}, []);
useHotkeys('shift+enter', handleRun, {preventDefault: true, enableOnFormTags: true}, [handleRun]);
useHotkeys('meta+shift+enter', handleSubmit, {preventDefault: true, enableOnFormTags: true}, [handleSubmit]);
```

And I wanted to add another one in a different file:
```tsx
useHotkeys('meta+shift+ArrowRight', moveToNextPage, {preventDefault: true, enableOnFormTags: true}, [moveToNextPage]);
```

If I add this to the same initial file it works fine, but as soon as I add it to another file the whole thing breaks.

**Comment by nicksrandall** [comment]
I have a PR in (#839) to fix this. I personally needed the fix ASAP so I published a fork on NPM with the changes: `@nicksrandall/react-hotkeys-hook`

**Comment by JohannesKlauss** [comment]
@nicksrandall Thanks for the PR! I added a few comments. Maybe you can have a look at it before we can merge it

**Comment by JohannesKlauss** [comment]
Hey all, based on @nicksrandall work, I have released version [4.0.4-1](https://www.npmjs.com/package/react-hotkeys-hook/v/4.0.4-1)

Please check if the issue still persists

**Comment by JohannesKlauss** [comment]
Hey,
so with https://www.npmjs.com/package/react-hotkeys-hook/v/4.0.4-5 the problems should be gone, please check this version out and give me feedback if the issue is solved.

**Comment by tmcw** [comment]
👍 Can confirm, 4.0.4-5 works in my setup, thanks!

**Comment by lukebowerman** [comment]
Ditto. Working for me as well. Thank you

**Comment by MartinXPN** [comment]
Can confirm for `4.0.4-5`, works like a charm. Thanks @JohannesKlauss @nicksrandall !

**Comment by JohannesKlauss** [comment]
I published version 4.0.4 as a stable patch. Thanks for the patience and help everybody
